### PR TITLE
website: docs for labels

### PR DESCRIPTION
**Pull request by mitchellh**

### CODE REVIEW

**Review by briancain — APPROVED** [review]

File: website/content/docs/lifecycle/labels.mdx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+that will be applied to the build step only for this invocation.
+
+```shell-session
+$ waypoint build -label a=1 -label b=2
+```
+
+### Project and App Lables
+
+Labels can also be specified at both the project and app scope. These
+lable are inherited by all operations within the project and application.
```
Comment (catsby) [inline website/content/docs/lifecycle/labels.mdx:50]:
```suggestion
labels are inherited by all operations within the project and application.
```

File: website/content/docs/lifecycle/labels.mdx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+"label selectors." Label selectors are
+[boolean expressions](https://en.wikipedia.org/wiki/Boolean_expression)
+(result in a "true" or "false"). An example of where filtering can be
+used is [application configuration](/docs/app-config/scope#label-scoping).
+
+The [full label selector syntax](#label-selector-syntax) is documented below.
+
+-> **Note:** Labels have powerful but limited functionality at present.
+We plan for more features to utilize labels in the future but expect their
+usage beyond simple informational tasks to remain for advanced concepts.
```
Comment (catsby) [inline website/content/docs/lifecycle/labels.mdx:108]:
```suggestion
usage beyond simple informational tasks to remain advanced concepts.
```

Nit: I'm not totally sure removing `for` here makes sense, but it reads better to me. Take or leave!

File: website/content/docs/lifecycle/labels.mdx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+Operation labels can also be specified using the CLI during any operation
+by specifying the `-label` flag. This flag can be repeated multiple times
+to specify multiple labels. The example below specifies a couple labels
+that will be applied to the build step only for this invocation.
+
+```shell-session
+$ waypoint build -label a=1 -label b=2
+```
+
+### Project and App Lables
```
Comment (catsby) [inline website/content/docs/lifecycle/labels.mdx:47]:
```suggestion
### Project and App Labels
```

**Review by catsby — APPROVED** [review]
Thanks for adding this! LGTM just minor nits

**Pull request merged**